src/components/input/input_text.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import classNamesDedupe from 'classnames/dedupe';
+
+const INPUT_TEXT_CLASSNAME = 'input-text';
+const INPUT_TEXT_ERROR_CLASSNAME = 'input-text--error';
+
+interface InputTextProps extends React.InputHTMLAttributes<HTMLInputElement> {
+    hasError?: boolean;
+    // prevent creating radio buttons and checkboxes
+    type?: 'text' | 'password' | 'email' | 'url' | 'number';
```

Comment:
Although HTML stuffs number input functionality into the standard INPUT element, inputting a number is somewhat different than normal text. For example, numbers can have min/max constraints, could also be replaced with a slider, and also often have small buttons for increment/decrement it.

Number input should probably be a different component. A password component with a "reveal" button built in, may also make more sense as a separate component.

Reply:
If it were ever needed, this low level component would allow you to generate an `<input type="number" min="0" max="100" />`, a slide would be `type="range"` but I don't think we'd need that kind of input anytime soon.

As for password entry with a reveal button or something, that's a component that would use this one.

My issue was that I needed an `<input class="I-don-t-want-to-have-to-think-about-this">` but basically could not have it. @ianpaschal I think this might address at least one of your concerns as well.